src/SignalR.Orleans/HostingExtensions.cs

```diff
@@ -47,27 +36,38 @@ public static ISiloHostBuilder UseSignalR(this ISiloHostBuilder builder, Action<
 
             cfg.ConfigureBuilder?.Invoke(builder, new HostBuilderConfig());
 
-            try
-            {
-                builder.AddMemoryGrainStorage(Constants.PUBSUB_PROVIDER);
-            }
-            catch
-            {
-                /** PubSubStore was already added. Do nothing. **/
-            }
-
-            try
-            {
-                builder.AddMemoryGrainStorage(Constants.STORAGE_PROVIDER);
-            }
-            catch
-            {
-                /** Grain storage provider was already added. Do nothing. **/
-            }
+            builder.ConfigureServices(services => services.AddSingleton<IConfigurationValidator, SignalRConfigurationValidator>());
 
             return builder
                 .AddSimpleMessageStreamProvider(Constants.STREAM_PROVIDER, opt => opt.FireAndForgetDelivery = cfg.UseFireAndForgetDelivery)
                 .ConfigureApplicationParts(parts => parts.AddApplicationPart(typeof(ClientGrain).Assembly).WithReferences());
         }
     }
+
+    internal class SignalRConfigurationValidator : IConfigurationValidator
+    {
+        private readonly IServiceProvider _sp;
+        private readonly ILogger _logger;
+
+        public SignalRConfigurationValidator(IServiceProvider serviceProvider)
+        {
+            this._logger = serviceProvider.GetRequiredService<ILoggerFactory>().CreateLogger<SignalRConfigurationValidator>();
```

Comment:
i guess you can change this to `ILogger<SignalRConfigurationValidator> logger` in the ctor